### PR TITLE
Feature/elementwise functions out keyword

### PR DESCRIPTION
**Pull request by vtavana**
Adding out keyword for elementwise functions

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to an issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [X] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: dpctl/tests/elementwise/test_add.py [hunk 1]
```diff
@@ -165,3 +166,79 @@ def __sycl_usm_array_interface__(self):
...
+    r = dpt.add(ar1, ar2)
+
+    y = dpt.zeros_like(ar1, dtype=r.dtype)
+    dpt.add(ar1, ar2, y)
+
+    assert np.array_equal(dpt.asnumpy(r), dpt.asnumpy(y))
+
+
+def test_add_errors():
+    ar1 = dpt.ones(2, dtype="float32", device="gpu")
```
Comment (oleksandr-pavlyk) [inline dpctl/tests/elementwise/test_add.py:191]: Gpu may mot be available, and this configuration must be handled. Does this test require a gpu though?

File: dpctl/tests/elementwise/test_add.py [hunk 2]
```diff
@@ -165,3 +166,79 @@ def __sycl_usm_array_interface__(self):
...
+    ar2 = dpt.ones_like(ar1, dtype="int32", device="gpu")
+    y = dpt.empty_like(ar1, device="cpu")
```
Comment (oleksandr-pavlyk) [inline dpctl/tests/elementwise/test_add.py:193]: Same here. A cpu may be unavailable.

**Review by oleksandr-pavlyk — CHANGES_REQUESTED** [review]

**Comment by oleksandr-pavlyk** [comment]
```
In [1]: import dpctl.tensor as dpt, numpy as np

In [2]: m = dpt.ones((10, 10), dtype="f4")

In [3]: v = dpt.ones(10, dtype="i4")

In [4]: r = dpt.empty_like(m)

In [5]: dpt.add(m, v, out=r)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [5], in <cell line: 1>()
----> 1 dpt.add(m, v, out=r)

File ~/repos/dpctl/dpctl/tensor/_elementwise_common.py:406, in BinaryElementwiseFunc.__call__(self, o1, o2, out, order)
    401     raise TypeError(
    402         f"output array must be of usm_ndarray type, got {type(out)}"
    403     )
    405 if out.shape != o1_shape or out.shape != o2_shape:
--> 406     raise TypeError(
    407         "The shape of input and output arrays are inconsistent."
    408         f"Expected output shape is {o1_shape}, got {out.shape}"
    409     )
    411 if ti._array_overlap(o1, out) or ti._array_overlap(o2, out):
    412     raise TypeError("Input and output arrays have memory overlap")

TypeError: The shape of input and output arrays are inconsistent.Expected output shape is (10, 10), got (10, 10)
```

**Comment by oleksandr-pavlyk** [comment]
Changes in this PR lead to a drop in coverage (84.036% in the target branch vs.  83.897% in this branch). Since your changes only modify `_elementwise_common.py`, please use `pytest-cov` to identify gaps and add required tests.

**Comment by vtavana** [comment]
> Changes in this PR lead to a drop in coverage (84.036% in the target branch vs. 83.897% in this branch). Since your changes only modify `_elementwise_common.py`, please use `pytest-cov` to identify gaps and add required tests.

I added new tests and now the coverage for `_elementwise_common.py` is 91% compared to 89%. All the lines that are added in this PR are tested now.

**after** adding new tests
dpctl/tensor/_elementwise_common.py         300     26    176     15    **91%**   83, 151, 161, 185-186, 229, 239, 243, 254, 259, 277, 315, 318, 341, 361, 372-373, 390, 475-477, 487, 517-519, 529

**before** adding new tests:
dpctl/tensor/_elementwise_common.py         300     30    176     20    **89%**   83, 95, 119->124, 151, 161, 185-186, 229, 239, 243, 254, 259, 277, 315, 318, 341, 361, 372-373, 390, 459, 475-477, 487, 496, 517-519, 529, 538

**Comment by oleksandr-pavlyk** [comment]
Tests fail on my WSL where GPU does not support fp64.

Specifically `test_add_dtype_matrix` where NumPy returns "double" and `test_add_dtype_error` which attempts to use `"f8"` regardless target device capabilities.

File: dpctl/tests/elementwise/test_abs.py [hunk 3]
```diff
@@ -22,9 +22,17 @@ def test_abs_out_type(dtype):
             np.dtype("c16"): np.dtype("f8"),
         }
         assert dpt.abs(X).dtype == type_map[arg_dt]
+
+        out = dpt.empty_like(X, dtype=type_map[arg_dt])
+        dpt.abs(X, out)
```
Comment (oleksandr-pavlyk) [inline dpctl/tests/elementwise/test_abs.py:27]: I think `out` should not be allowed to be a positional argument in array API. Could you please change it to a keyword argument in the test suite?

Comment (vtavana) [reply dpctl/tests/elementwise/test_abs.py:27]: sure!

**Review by oleksandr-pavlyk — APPROVED** [review]
Thank you @vtavana ! Please merge when ready

**Pull request merged**